libcnb-cargo/src/main.rs

```diff
@@ -223,3 +379,332 @@ fn calculate_dir_size(path: impl AsRef<Path>) -> io::Result<u64> {
 
     Ok(size_in_bytes)
 }
+
+fn get_buildpack_workspace() -> BuildpackWorkspace {
+    let workspace_root = get_workspace_root();
+    let workspace_cargo_metadata = get_cargo_metadata(&workspace_root);
+    let workspace_target_directory = workspace_cargo_metadata
+        .target_directory
+        .into_std_path_buf();
+    BuildpackWorkspace {
+        root: workspace_root,
+        target_dir: workspace_target_directory,
+    }
+}
+
+#[allow(clippy::too_many_lines)]
+fn get_buildpack_projects(
+    buildpack_workspace: &BuildpackWorkspace,
+    args: &PackageArgs,
+) -> Vec<BuildpackProject> {
+    let mut buildpack_projects: Vec<BuildpackProject> = vec![];
+
+    for buildpack_dir in find_buildpack_directories(buildpack_workspace) {
+        let buildpack_data = match read_buildpack_data(&buildpack_dir) {
+            Ok(buildpack_data) => buildpack_data,
+            Err(error) => {
+                warn(formatdoc! { "
+                    Ignoring buildpack project from {} 
+
+                    To include this project, please verify that the `buildpack.toml` file:
+                    • is readable
+                    • contains valid buildpack metadata
+
+                    Error: {:#?}
+                ", &buildpack_dir.to_string_lossy(), error });
+                continue;
+            }
+        };
+
+        let buildpackage_data = if buildpack_dir.join("package.toml").exists() {
+            match read_buildpackage_data(&buildpack_dir) {
+                Ok(buildpackage_data) => Some(buildpackage_data),
+                Err(error) => {
+                    warn(formatdoc! { "
+                        Ignoring buildpack project from {} 
+    
+                        To include this project, please verify that the `package.toml` file:
+                        • is readable
+                        • contains valid buildpackagage metadata
+    
+                        Error: {:#?}
+                    ", &buildpack_dir.to_string_lossy(), error });
+                    continue;
+                }
+            }
+        } else {
+            None
+        };
+
+        let buildpack_target_dir = if is_legacy_buildpack_project(&buildpack_dir) {
+            buildpack_dir.clone()
```

Comment:
Why is that handling here? Feels like a hack to make things work - should this function even run for "legacy" buildpack packages?